forms/Form.php

```diff
@@ -286,6 +286,9 @@ public function __construct($controller, $name, FieldList $fields, FieldList $ac
 		$this->securityToken = ($securityEnabled) ? new SecurityToken() : new NullSecurityToken();
 
 		$this->setupDefaultClasses();
+
+		// ensure security token is included in the field list
+		$this->ensureSecurityIDField();
```

Comment:
This changes the behaviour of `Form->disableSecurityToken()`. By default, a security token is required in forms, hence the logic would add the field (via `ensureSecurityIDField()`). If it's later disabled on the form instance, you would now also need to remove the field. Which is why we deferred adding the field until the last moment (since `getExtraFields()` is likely called as part of render). Admittedly it's fragile, but your pull request is an API change which makes the logic more consistent in one place, to make it less consistent in another.